zebra-network/src/peer/client.rs

```diff
@@ -63,9 +143,15 @@ impl Service<Request> for Client {
         // request.
         let span = tracing::Span::current();
 
-        match self.server_tx.try_send(ClientRequest { request, span, tx }) {
+        match self.server_tx.try_send(ClientRequest {
+            request,
+            span,
+            tx: tx.into(),
+        }) {
             Err(e) => {
                 if e.is_disconnected() {
+                    let ClientRequest { tx, .. } = e.into_inner();
+                    let _ = tx.send(Err(PeerError::ConnectionClosed.into()));
```

Comment:
This change won't be required with the new design, but it's still an improvement over the old code.